farmbot_ext/lib/farmbot_ext/amqp/connection_worker.ex

```diff
@@ -91,4 +96,8 @@ defmodule FarmbotExt.AMQP.ConnectionWorker do
 
     AMQP.Connection.open(opts)
   end
+
+  def maybe_connect(jwt_bot_claim) do
+    FarmbotExt.fetch_impl!(@me, :network_impl).maybe_connect(jwt_bot_claim)
```

Comment:
again, i like this idea, but i think it should be implemented slightly differently. The calling module shouldn't really be doing implementation lookups.

Reply:
Whoops! i didn't realize what module this was in. This call is ok, but i still believe it needs a bit more tweaking.